src/U_polynomials.jl

```diff
@@ -36,12 +39,47 @@ function besseljy_debye(v, x)
     p = v / vs
     p2  = v^2 / vmx
 
-    Uk_Jn, Uk_Yn = Uk_poly_Jn(p, v, p2, x)
-
+    Uk_Jn, Uk_Yn = Uk_poly_Jn(p, v, p2, T(x))
     return coef_Jn * Uk_Jn, coef_Yn * Uk_Yn
 end
 
-besseljy_debye_cutoff(nu, x) = nu > 2.0 + 1.00035*x + Base.Math._approx_cbrt(Float64(302.681)*x) && nu > 15
+# Cutoffs for besseljy_debye expansions
+# regions where the debye expansions for large orders v > x are valid
+# determined by fitting a curve a + bx + (cx)^(1/3) to where debye expansions provide desired precision
+
+# Float32
+besseljy_debye_fit(x::Float32) =  2.5f0 + 1.00035f0*x + Base.Math._approx_cbrt(360.0f0*x)
+besseljy_debye_cutoff(nu, x::Float32) = nu > besseljy_debye_fit(x) && nu > 6
+
+# Float64
+besseljy_debye_fit(x::Float64) = 2.0 + 1.00035*x + Base.Math._approx_cbrt(302.681*x)
+besseljy_debye_cutoff(nu, x::Float64) = nu > besseljy_debye_fit(x) && nu > 15
+
+# Float128 - provide roughly ~1e-35 precision
+besseljy_debye_fit(x) = 16.0 + 1.0012*x + Base.Math._approx_cbrt(Float64(27.91*x))
+besseljy_debye_cutoff(nu, x) = nu > besseljy_debye_fit(x) && nu > 40
+
+#####
+##### Debye large order expansion coefficients
+#####
+
+function Uk_poly_Jn(p, v, p2, x::Float64)
+    if v > 5.0 + 1.00033*x + Base.Math._approx_cbrt(1427.61*x)
```

Comment:
It's probably better to move the multiplication outside the cbrt.

Reply:
You mean like `5.0 + 1.00033*x +11.45 * Base.Math._approx_cbrt(x)`?